src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  constructor(name, health) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+  bite(animal) {
+    if (animal.hasOwnProperty('hidden')) {
+      if (!animal.hidden) {
+        animal.health -= 50;
+      }
+
+      if (animal.health <= 0) {
+        Animal.alive = Animal.alive.filter(
+          (aliveAnimal) => aliveAnimal !== animal,
+        );
+      }
+    }
```

Comment:
[CODE QUALITY] - The `bite` method should check if the `animal` parameter is an instance of `Herbivore` and not hidden, rather than checking for the existence of a 'hidden' property. This ensures that the method adheres to the class structure and does not rely on property checks that might not be relevant for other types of objects. Consider using `instanceof` to verify the object's class.